### ISSUE TITLE
Unable to install nio4r gem via Vagrant plugin

### ISSUE THREAD

**Issue by just3ws** [hashicorp/vagrant#2402]
Hi,

TL;DR The nio4r gem builds fine on my Ruby 1.9.3 but fails to build from Vagrant's Ruby interpreter.

There seems to be an issue with Xcode >=4.6.3 on Mountain Lion being unable to compile some gems with native code using the Vagrant >= 1.3.4 embedded Ruby, in this case nio4r. 

This is a continuation of an issue started over on the vagrant-berkshelf repository. 

https://github.com/RiotGames/vagrant-berkshelf/issues/89

One other vagrant-berkshelf user, @zepheiryan dug into the issue a little deeper, from the thread on the vagrant-berkshelf issue.

> I'm having the same problem with about the same environment, except I have a full XCode install.
> 
> Looking at the files the error messages point to, I see the conflict is that OS X defines structs timeval and timespec, but the parameters to tell Ruby that those aren't missing aren't provided in nio4r's extconf.rb (it should define HAVE_STRUCT_TIMESPEC and HAVE_STRUCT_TIMEVAL based on the presence of sys/_structs.h, perhaps in a platform-dependent manner; it does not).
> 
> I "solved" this by temporarily commenting out the struct definitions of Vagrant's embedded Ruby's missing.h. I also had to make sure my ARCHFLAGS was unset for my environment because the embedded Ruby does not provide headers for x86_64-darwin12.3.0, only universal. The install succeeded after making those changes.
> 
> I have no qualms about doing this locally just to get this working since Ruby development is low on my activities list, but I'm guessing as a real solution, this is an issue to bring up with the nio4r team. I'll file an issue there if that makes sense to others.

I've also submitted an error directly on the nio4r (the gem throwing the exception) but since this was because Vagrant requires using a custom build of Ruby I'd also submit the issue here.

https://github.com/celluloid/nio4r/issues/35
### Full output of the error trace below:

unknown28cfe9162235 :: ~/src/vgrnt ‹master*› % vagrant plugin install vagrant-berkshelf                                                                                                                                                   1 ↵
Installing the 'vagrant-berkshelf' plugin. This can take a few minutes...
/Applications/Vagrant/embedded/lib/ruby/1.9.1/rubygems/installer.rb:562:in `rescue in block in build_extensions': ERROR: Failed to build gem native extension. (Gem::Installer::ExtensionBuildError)

```
    /Applications/Vagrant/embedded/bin/ruby extconf.rb
```

checking for rb_thread_blocking_region()... **\* extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of
necessary libraries and/or headers.  Check the mkmf.log file for more
details.  You may need configuration options.

Provided configuration options:
    --with-opt-dir
    --with-opt-include
    --without-opt-include=${opt-dir}/include
    --with-opt-lib
    --without-opt-lib=${opt-dir}/lib
    --with-make-prog
    --without-make-prog
    --srcdir=.
    --curdir
    --ruby=/Applications/Vagrant/embedded/bin/ruby
/Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:381:in `try_do': The compiler failed to generate an executable file. (RuntimeError)
You have to install development tools first.
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:455:in`block in try_link0'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/tmpdir.rb:83:in `mktmpdir'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:452:in`try_link0'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:476:in `try_link'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:619:in`try_func'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:894:in `block in have_func'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:790:in`block in checking_for'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:284:in `block (2 levels) in postpone'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:254:in`open'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:284:in `block in postpone'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:254:in`open'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:280:in `postpone'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:789:in`checking_for'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/mkmf.rb:893:in `have_func'
    from extconf.rb:3:in`<main>'

Gem files will remain installed in /Users/Michael/.vagrant.d/gems/gems/nio4r-0.5.0 for inspection.
Results logged to /Users/Michael/.vagrant.d/gems/gems/nio4r-0.5.0/ext/nio4r/gem_make.out
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/rubygems/installer.rb:540:in `block in build_extensions'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/rubygems/installer.rb:515:in`each'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/rubygems/installer.rb:515:in `build_extensions'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/rubygems/installer.rb:180:in`install'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/rubygems/dependency_installer.rb:297:in `block in install'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/rubygems/dependency_installer.rb:270:in`each'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/rubygems/dependency_installer.rb:270:in `each_with_index'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/rubygems/dependency_installer.rb:270:in`install'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/commands/plugin/action/install_gem.rb:49:in `block in call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/commands/plugin/gem_helper.rb:42:in`block in with_environment'
    from /Applications/Vagrant/embedded/lib/ruby/1.9.1/rubygems/user_interaction.rb:40:in `use_ui'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/commands/plugin/gem_helper.rb:41:in`with_environment'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/commands/plugin/action/install_gem.rb:39:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in`call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/commands/plugin/action/bundler_check.rb:20:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/warden.rb:34:in`call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/builder.rb:116:in `call'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/runner.rb:61:in`block in run'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/util/busy.rb:19:in `busy'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/action/runner.rb:61:in`run'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/commands/plugin/command/base.rb:17:in `action'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/commands/plugin/command/install.rb:27:in`execute'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/plugins/commands/plugin/command/root.rb:52:in `execute'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/cli.rb:38:in`execute'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/lib/vagrant/environment.rb:478:in `cli'
    from /Applications/Vagrant/embedded/gems/gems/vagrant-1.3.5/bin/vagrant:96:in`<top (required)>'
    from /Applications/Vagrant/bin/../embedded/gems/bin/vagrant:23:in `load'
    from /Applications/Vagrant/bin/../embedded/gems/bin/vagrant:23:in`<main>'
